Sources/GRPC/GRPCClientConnection.swift

```diff
@@ -171,7 +153,163 @@ open class GRPCClientConnection {
+  public struct TLSConfiguration {
+    /// The SSL context to use.
+    public var sslContext: NIOSSLContext
+    /// Value to use for TLS SNI extension; this must not be an IP address.
+    public var hostnameOverride: String?
+
+    public init(sslContext: NIOSSLContext, hostnameOverride: String? = nil) {
+      self.sslContext = sslContext
+      self.hostnameOverride = hostnameOverride
+    }
+  }
+}
+
+// MARK:- Configuration helpers/extensions
+
+fileprivate extension ClientBootstrap {
+  /// Connect to the given connection target.
+  ///
+  /// - Parameter target: The target to connect to.
+  func connct(to target: ConnectionTarget) -> EventLoopFuture<Channel> {
+    switch target {
+    case .hostAndPort(let host, let port):
+      return self.connect(host: host, port: port)
+
+    case .unixDomainSocket(let path):
+      return self.connect(unixDomainSocketPath: path)
+
+    case .socketAddress(let address):
+      return self.connect(to: address)
+    }
+  }
+}
+
+fileprivate extension Channel {
+  /// Configure the channel with TLS.
+  ///
+  /// This function adds two handlers to the pipeline: the `NIOSSLClientHandler` to handle TLS, and
+  /// the `GRPCTLSVerificationHandler` which verifies that a successful handshake was completed.
+  ///
+  /// - Parameter configuration: The configuration to configure the channel with.
+  /// - Parameter errorDelegate: The error delegate to use for the TLS verification handler.
+  func configureTLS(
+    _ configuration: GRPCClientConnection.TLSConfiguration,
+    errorDelegate: ClientErrorDelegate?
+    ) -> EventLoopFuture<Void> {
+    do {
+      let sslClientHandler = try NIOSSLClientHandler(
+        context: configuration.sslContext,
+        serverHostname: configuration.hostnameOverride)
+
+      let verificationHandler = GRPCTLSVerificationHandler(errorDelegate: errorDelegate)
+      return self.pipeline.addHandlers(sslClientHandler, verificationHandler)
+    } catch {
+      return self.eventLoop.makeFailedFuture(error)
+    }
+  }
+}
+
+// MARK:- Legacy APIs
+
+extension GRPCClientConnection {
+  /// Starts a connection to the given host and port.
+  ///
+  /// - Parameters:
+  ///   - host: Host to connect to.
+  ///   - port: Port on the host to connect to.
+  ///   - eventLoopGroup: Event loop group to run the connection on.
+  ///   - errorDelegate: An error delegate which is called when errors are caught. Provided
+  ///       delegates **must not maintain a strong reference to this `GRPCClientConnection`**. Doing
+  ///       so will cause a retain cycle. Defaults to a delegate which logs errors in debug builds
+  ///       only.
+  ///   - tlsMode: How TLS should be configured for this connection.
+  ///   - hostOverride: Value to use for TLS SNI extension; this must not be an IP address. Ignored
+  ///       if `tlsMode` is `.none`.
+  /// - Returns: A future which will be fulfilled with a connection to the remote peer.
+  public static func start(
```

Comment:
Should we mark this as deprecated?

Reply:
I was going to remove it in the next PR so didn't bother :)